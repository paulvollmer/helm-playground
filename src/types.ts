declare global {
  interface Window {
    helmRender: (templates: string, values: string, chart: string, settings: string) => helmRenderReturn
    helmDefaultCapabilities: () => SettingsData
  }
}

export type helmRenderReturn = {
  error?: helmRenderReturnError
  result?: string
}

export type helmRenderReturnError = {
  kind: string
  file: string
  line: number
  message: string
}

export type Sources = {
  [key: string]: string
}

export type SettingsData = {
  release: SettingsRelease
  kubeVersion: SettingsKubeVersion
  helmVersion: SettingsHelmVersion
}

export type SettingsRelease = {
  name: string
  namespace: string
  revision: string
  isUpgrade: string
  isInstall: string
  service: string
}

export type SettingsKubeVersion = {
  version: string
  major: string
  minor: string
}

export type SettingsHelmVersion = {
  version: string
  gitCommit: string
  gitTreeState: string
  goVersion: string
}
