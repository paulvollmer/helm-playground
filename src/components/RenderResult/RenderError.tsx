import React from 'react'
import { Typography } from '@mui/material'
import { HelmRenderReturnError } from '../../types'

export type RenderErrorProps = {
  error: HelmRenderReturnError
}

export const RenderError = (props: RenderErrorProps): JSX.Element => {
  return (
    <div>
      <Typography variant="subtitle1">Error Kind: {props.error.kind}</Typography>
      <Typography variant="h6">{props.error.message}</Typography>
      <Typography variant="body1">
        File: {props.error.file}
        <br />
        {props.error.line ? `Line: ${props.error.line}` : ''}
      </Typography>
    </div>
  )
}
