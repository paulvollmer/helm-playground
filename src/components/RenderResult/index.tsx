import React from "react";
import AceEditor from "react-ace";
import "ace-builds/src-noconflict/mode-yaml";
import {helmRenderReturn, helmRenderReturnError} from "../../types";
import {Typography} from "@material-ui/core";

type RenderResultProps = {
    data: helmRenderReturn
}

/**
 * call the helmRender function and dispay the result
 */
const RenderResult = (props: RenderResultProps) => {
    return (
        <>
            {props.data.error ? (
                <RenderError error={props.data.error}/>
            ) : (
                <>
                    <Typography variant="subtitle1">Render Output</Typography>
                    <AceEditor
                        mode="yaml"
                        theme="github"
                        name="editor"
                        width="100%"
                        height="80vh"
                        value={props.data.result}
                        editorProps={{$blockScrolling: true}}
                    />
                </>
            )}
        </>
    )
}


type RenderErrorProps = {
    error: helmRenderReturnError;
}

const RenderError = (props: RenderErrorProps) => {
    return (
        <div>
            <Typography variant="subtitle1">
                Error Kind: {props.error.kind}
            </Typography>
            <Typography variant="h6">
                File: {props.error.file} {props.error.line ? (`Line: ${props.error.line}`): ""}
            </Typography>
            <Typography variant="body1">
                {props.error.message}
            </Typography>
        </div>
    )
}

export default RenderResult
