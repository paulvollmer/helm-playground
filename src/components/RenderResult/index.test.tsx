import React from 'react'
import TestRenderer from 'react-test-renderer'
import RenderResult from './index'
import { HelmRenderReturn } from '../../types'

test('renders RenderResult with result', () => {
  const testData: HelmRenderReturn = { result: 'test-result' }
  const tree = TestRenderer.create(<RenderResult data={testData} />).toJSON()
  expect(tree).toMatchSnapshot()
})

test('renders RenderResult with error', () => {
  const testData: HelmRenderReturn = {
    error: {
      kind: 'test-kind',
      file: 'test-file',
      line: 1,
      message: 'test-message',
    },
  }
  const tree = TestRenderer.create(<RenderResult data={testData} />).toJSON()
  expect(tree).toMatchSnapshot()
})
