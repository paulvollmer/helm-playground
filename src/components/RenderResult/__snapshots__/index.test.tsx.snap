// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RenderResult render with result 1`] = `
Array [
  <h6
    className="MuiTypography-root MuiTypography-subtitle1 css-10wpov9-MuiTypography-root"
  >
    Render Output
  </h6>,
  <div
    id="editor"
    style={
      Object {
        "height": "calc(100vh - 110px)",
        "width": "100%",
      }
    }
  />,
]
`;

exports[`RenderResult renders with error 1`] = `
<div>
  <h6
    className="MuiTypography-root MuiTypography-subtitle1 css-10wpov9-MuiTypography-root"
  >
    Error Kind: 
    test-kind
  </h6>
  <h6
    className="MuiTypography-root MuiTypography-h6 css-2ulfj5-MuiTypography-root"
  >
    test-message
  </h6>
  <p
    className="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
  >
    File: 
    test-file
    <br />
    Line: 1
  </p>
</div>
`;
